### PR TITLE
Added functionality for back button to not render if on page after lo…

### PR DESCRIPTION
**Pull request by brianseek**
…ading screen.

## Description
The back button should not exist on the first page after logging into the day-of check-in app. It isn't rendered on the demographics page at all but if this page is skipped, it is included in the next page. This change adds the ability for the back button to look back in the router and decide if it should be rendered. 

This is not an issue on pre-check-in since we have the introduction page.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#35247

## Testing done
Updated unit test for back button to include checking if it should render.

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/check-in/components/BackButton.jsx [hunk 1]
```diff
@@ -4,20 +4,31 @@ import PropTypes from 'prop-types';
...
+import { useFormRouting } from '../hooks/useFormRouting';
 
 const BackButton = props => {
-  const { action, path } = props;
+  const { action, router } = props;
+  const { getCurrentPageFromRouter, pages } = useFormRouting(router);
+
+  const currentPage = getCurrentPageFromRouter();
+  const positionInForm = pages.indexOf(currentPage);
+  const previousPage = pages[positionInForm - 1];
```
Comment (mdewey) [inline src/applications/check-in/components/BackButton.jsx:15]: Could probably do a new function in the `useFormRouting` called `getPreviousPage`. I can see this being useful for the new routing stuff we that might be coming

File: src/applications/check-in/components/BackButton.jsx [hunk 2]
```diff
@@ -4,20 +4,31 @@ import PropTypes from 'prop-types';
...
-        fromPage: path,
+        fromPage: currentPage,
       });
       action();
     },
-    [path, action],
+    [currentPage, action],
   );
+
+  if (previousPage === 'loading-appointments') {
```
Comment (mdewey) [inline src/applications/check-in/components/BackButton.jsx:29]: Use the `URLS` object for the `'loading-appointments'` URL comparison

**Review by mdewey — CHANGES_REQUESTED** [review]
Overall good work,  two small changes.

File: src/applications/check-in/components/BackButton.jsx (hunk 1 above)
Comment (brianseek) [reply src/applications/check-in/components/BackButton.jsx:15]: I feel like I was going to do that, but then after coming back to it after a day of meetings, I spaced on it. Thanks for flagging that!

File: src/applications/check-in/components/BackButton.jsx (hunk 2 above)
Comment (brianseek) [reply src/applications/check-in/components/BackButton.jsx:29]: great idea!

**Review by mdewey — APPROVED** [review]
Nice! There is 1 linting error, but other than that it looks good

**Pull request merged**